### ISSUE TITLE
mingw-w64-ffmpeg does not build

### ISSUE THREAD

**Issue by ferkulat** [msys2/MINGW-packages#20891]
### Description / Steps to reproduce the issue

Hello, I try to build mingw-w64-ffmpeg
1. `pacman -Syu`
2. `git clone  https://github.com/msys2/MINGW-packages.git`
3. `cd MINGW-packages/mingw-w64-ffmpeg`
4. `git checkout d9f875f711047a056376a567e0f6b1cfae88f3c0` ( current master by the time of writing)
5. `makepkg -rs --skippgpcheck` (failed gpg check is an additional issue)

### Expected behavior

should build package

### Actual behavior

I get a build failure
```
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c: In function 'nvenc_setup_hevc_config':
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1373:9: error: 'NV_ENC_CONFIG_HEVC' has no member named 'pixelBitDepthMinus8'
 1373 |     hevc->pixelBitDepthMinus8 = IS_10BIT(ctx->data_pix_fmt) ? 2 : 0;
      |         ^~
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c: In function 'nvenc_setup_av1_config':CC       libavcodec/pthread_frame.o

D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1458:8: error: 'NV_ENC_CONFIG_AV1' has no member named 'inputPixelBitDepthMinus8'
 1458 |     av1->inputPixelBitDepthMinus8 = IS_10BIT(ctx->data_pix_fmt) ? 2 : 0;
      |        ^~
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1459:8: error: 'NV_ENC_CONFIG_AV1' has no member named 'pixelBitDepthMinus8'
 1459 |     av1->pixelBitDepthMinus8 = (IS_10BIT(ctx->data_pix_fmt) || ctx->highbitdepth) ? 2 : 0;
      |        ^~
CC      libavcodec/pthread_slice.o
CC      libavcodec/ptx.o
CC      libavcodec/qcelpdec.o
CC      libavcodec/qdm2.o
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c: In function 'nvenc_map_buffer_format':
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1692:16: error: 'NV_ENC_BUFFER_FORMAT_YV12_PLCC        libavcodec/qdmc.o
' undeclared (first use in this function); did you mean 'NV_ENC_BUFFER_FORMAT_YV12'?
 1692 |         return NV_ENC_BUFFER_FORMAT_YV12_PL;
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~CC   libavcodec/qdrw.o
CC      libavcodec/qoi_parser.o

      |                NV_ENC_BUFFER_FORMAT_YV12
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1692:16: note: each undeclared identifier is reported only once for each function it appears in
CC      libavcodec/qoidec.o
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1694:16: error: 'NV_ENC_BUFFER_FORMAT_NV12_PL' undeclared (first use in this function); did you mean 'NV_ENC_BUFFER_FORMAT_NV12'?
 1694 |         return NV_ENC_BUFFER_FORMAT_NV12_PL;
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                NV_ENC_BUFFER_FORMAT_NV12
CC      libavcodec/qpeg.o
CC      libavcodec/qoienc.o
CC      libavcodec/qpeldsp.o
CC      libavcodec/qsv.o
D:/devel/vms-mingw-packages/mingw-w64-ffmpeg/src/ffmpeg-6.1.1/libavcodec/nvenc.c:1700:16: error: 'NV_ENC_BUFFER_FORMAT_YUV444_PL' undeclared (first use in this function); did you mean 'NV_ENC_BUFFER_FORMAT_YUV444'?
 1700 |         return NV_ENC_BUFFER_FORMAT_YUV444_PL;
      |                CC       libavcodec/qsv_api.o
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~CC        libavcodec/qsvdec.o

      |                NV_ENC_BUFFER_FORMAT_YUV444
CC      libavcodec/qsvenc.o
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by Biswa96** [comment]
There is a compiler error due to gcc 14 update as following.

```
ffmpeg-6.1.1/libavutil/hwcontext_qsv.c:770:49: error: passing argument 2 of 'device->lpVtbl->GetDirect3D' from incompatible pointer type [-Wincompatible-pointer-types]
  770 |     hr = IDirect3DDevice9Ex_GetDirect3D(device, &d3d9ex);
      |                                                 ^~~~~~~
      |                                                 |
      |                                                 IDirect3D9Ex **
ffmpeg-6.1.1/libavutil/hwcontext_qsv.c:770:49: note: expected 'IDirect3D9 **' but argument is of type 'IDirect3D9Ex **'
```

I am not sure if that should be workaround with a simple type casting or changing the function.

**Comment by Biswa96** [comment]
I have reported the issue in upstream https://ffmpeg.org/pipermail/ffmpeg-user/2024-May/057946.html

**Comment by Biswa96** [comment]
Fixed by 87e8350700a8d6d7dd25a2023d247e2460b50fd9
